Buildings/Electrical/AC/ThreePhasesBalanced/Loads/MotorDrive/InductionMotors/SquirrelCage.mo

```diff
@@ -0,0 +1,166 @@
+            Real v_rms "RMS voltage";
+            Real pow_gap "Power gap";
+
+  Modelica.Units.SI.Angle theta_s
+    "Supply voltage phase angel";
+  Modelica.Units.SI.AngularVelocity omega
+    "Supply voltage angular frequency";
+
+  Modelica.Units.SI.Voltage v[:] = terminal.v
+    "Voltage vector";
+  Modelica.Units.SI.Current i[:] = terminal.i
+    "Current vector";
+  final Modelica.Blocks.Sources.RealExpression Vrms(y=v_rms) "RMS voltage"
+    annotation (Placement(transformation(extent={{-60,10},{-40,30}})));
+
+  final Modelica.Blocks.Sources.RealExpression fre(y=omega/(2*Modelica.Constants.pi))
+    "Supply voltage frequency"
+    annotation (Placement(transformation(extent={{-64,-10},{-44,10}})));
+
+  Modelica.Blocks.Interfaces.RealInput tau_m(unit="N.m")
+    "Load torque"
+    annotation (Placement(transformation(
+        extent={{-20,-20},{20,20}},
+        rotation=0,
+        origin={-120,-80}),
+        iconTransformation(
+        extent={{-20,-20},{20,20}},
+        rotation=0,
+        origin={-120,-80})));
+
+  Modelica.Blocks.Continuous.Integrator integrator
+    annotation (Placement(transformation(extent={{-2,60},{18,80}})));
+  Modelica.Blocks.Sources.RealExpression i_ds(y=torSpe.motMod.i_ds)
+    annotation (
+      Placement(transformation(extent={{-10,-12},{10,12}}, origin={30,40})));
+  Modelica.Blocks.Sources.RealExpression i_qs(y=torSpe.motMod.i_qs)
+    annotation (
+      Placement(transformation(extent={{-10,-12},{10,12}}, origin={30,24})));
+  Modelica.Blocks.Sources.RealExpression eleTor(y=torSpe.tau_e)
+    annotation (
+      Placement(transformation(extent={{-10,-12},{10,12}}, origin={-56,-58})));
+  Modelica.Blocks.Sources.RealExpression angFre(y=omega)
+    "Supply voltage angular frequency" annotation (Placement(transformation(
+          extent={{-10,-12},{10,12}}, origin={-32,-80})));
+  Modelica.Blocks.Sources.RealExpression angFre1(y=omega)
+    "Supply voltage angular frequency" annotation (Placement(transformation(
+          extent={{-10,-12},{10,12}}, origin={-30,70})));
+  Modelica.Mechanics.Rotational.Sources.Speed speed(exact=true)
+    "Speed connector"
+    annotation (Placement(transformation(extent={{64,-8},{80,8}})));
+  Modelica.Mechanics.Rotational.Interfaces.Flange_b shaft
+    "Mechanical connector"
+    annotation (Placement(transformation(extent={{90,-10},{110,10}})));
+
+  BaseClasses.MotorMachineInterface torSpe(
+    P=per.P,
+    J=per.J,
+    Lr=per.Lr,
+    Ls=per.Ls,
+    Rr=per.Rr,
+    Lm=per.Lm,
+    Rs=per.Rs) annotation (Placement(transformation(extent={{2,-6},{16,8}})));
+  BaseClasses.SpeedBlock speBlo(J=per.J, P=per.P)
+    annotation (Placement(transformation(extent={{-4,-76},{16,-56}})));
+  BaseClasses.CurrentBlock current_Block
+    annotation (Placement(transformation(extent={{66,28},{86,48}})));
+equation
+  // Assign values for motor model calculation from electrical interface
+  theta_s = PhaseSystem.thetaRef(terminal.theta) "phase angle";

```

Comment:
is it possible to create base class to do the calculations as showing in these equations? it seems that we have same pattern as in the `SquirrelCageDrive`?